src/std/misc/list.ss

```diff
@@ -285,3 +286,21 @@ package: std/misc
       (if (pair? cur)
 	(snoc (reverse! cur) acc)
 	acc)))))
+
+;; group consecutive elements of the list lst into a list-of-lists.
+;;
+;; Example:
+;;  (group [1 2 2 3 1 1]) => ((1) (2 2) (3) (1 1))
+(def (group lst (test equal?))
+  (def (helper)
+    (let loop ((lst lst) (buf []) (acc []))
+      (if (pair? lst)
+	(let (it (car lst))
```

Comment:
this could be written more concisely with match:
```
(let loop ((rest lst) (buf [])) (acc []))
  (match rest
    ([it . rest] ...)
    (else (reverse! ..))))
```